pkg/clusteragent/admission/mutate/agent_sidecar/profiles.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package agentsidecar
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	corev1 "k8s.io/api/core/v1"
+)
+
+////////////////////////////////
+//                            //
+//     Profiles Overrides     //
+//                            //
+////////////////////////////////
+
+// ProfileOverride represents environment variables and resource requirements overrides
+// It returns an error in case it fails to apply the profile overrides
+type ProfileOverride struct {
+	EnvVars              []corev1.EnvVar             `json:"env,omitempty"`
+	ResourceRequirements corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+// LoadSidecarProfiles returns the profile overrides provided by the user
+// It returns an error in case of miss-configuration or in case more than
+// one profile is configured
+func LoadSidecarProfiles() ([]ProfileOverride, error) {
```

Comment:
Why does this return an array if only one can be configured?

Reply:
In order to make it easy in the future to support multiple profiles.

For now, we support configuring at most 1 profile. I am still returning a slice to make migration easier in the future.